src/main/java/com/elastic/support/diagnostics/commands/RunKibanaQueries.java

```diff
@@ -0,0 +1,309 @@
+    */
+    private LocalSystem getLocalSystem(String osName, Boolean parseOperatingSystem) {
+        if (parseOperatingSystem == true) {
+            osName = SystemUtils.parseOperatingSystemName(SystemProperties.osName);
+        }
+        LocalSystem syscmd = new LocalSystem(osName);
+        ResourceCache.addSystemCommand(Constants.systemCommands, syscmd);
+        return syscmd;
+    }
+
+
+    /**
+    * CheckKibanaVersion (executed before) defined/set the context.elasticRestCalls.
+    * here we will loop and create a List of RestEntry with the queries/APIs that need to be executed
+    * You have two APIs that work differently and and may have or not many pages, so we use the getAllPages function
+    * runQueries called in this function create a json file for each RestEntry set on the 'queries' variable.
+    * Public function so we can test it.
+    *
+    * @param  RestClient client
+    * @param  DiagnosticContext context
+    * @return int
+    */
+    public int runBasicQueries(RestClient client, DiagnosticContext context) {
+
+        int totalRetries = 0;
+        List<RestEntry> queries = new ArrayList<>();
+        
+        for (Map.Entry<String, RestEntry> entry : context.elasticRestCalls.entrySet()) {
+
+            String actionName = entry.getValue().getName().toString();
+            if (actionName.equals("kibana_alerts") || actionName.equals("kibana_detection_engine_find")) {
+                getAllPages(client, queries, context.perPage, entry.getValue());
+            } else {
+                queries.add(entry.getValue());
+            }
+        }
+        totalRetries = runQueries(client, queries, context.tempDir, 0, 0);
+
+        return totalRetries;
+    }
+
+
+    /**
+    * On this function we will use the RestEntry action object to get the URL and execute the API one first time
+    * to get the total of events defined in Kibana for that API.
+    * Once we have the total of event we can calculate the number of pages/call we will need to execute
+    * then we call getNewEntryPage to create a new RestEntry for each page.
+    *
+    * @param  RestClient client
+    * @param  List<RestEntry> queries
+    * @param  double perPage - This is the number of docusment we reques to the API (set to 100 in execute func / or n in unit test)
+    * @param  RestEntry action
+    *
+    * @return         void
+    */
+    public void getAllPages(RestClient client, List<RestEntry> queries, double perPage, RestEntry action) {
+        // get the values needed to the pagination.
+        RestResult res = client.execQuery(String.format("%s?per_page=1", action.getUrl()));
+        if (! res.isValid()) {
+            throw new DiagnosticException( res.formatStatusMessage( "Could not retrieve Kibana API pagination - unable to continue."));
+        }
+        String result   = res.toString();
+        JsonNode root   = JsonYamlUtils.createJsonNodeFromString(result);
+        double total    = root.path("total").intValue();
```

Comment:
Why is this a `double`? It's explicitly an `int` from how it's parsed.

```suggestion
        int total       = root.path("total").intValue();
```